### PR TITLE
Feature/1603 mapping table filtering general

### PR DESCRIPTION
**Pull request by dk1844**
This is backend preparation for #1603. Most notably:
 - bson codec for `DataFrameFilter` has been added to mongo registry, otherwise, the SerDe would not work at all
 - Json SerDe for MT and DS tests with `DataFrameFilter`s being used 
 - DatasetAPI IntegTest extended to cover DataFrameFilter usage for Datasets.

# Release notes suggestion
 - DataFrameFilter BSON serialization fixed

### CODE REVIEW

**Review by AdrianOlosutean — DISMISSED** [review]
Looks good, will test tomorrow

File: dao/src/test/scala/za/co/absa/enceladus/dao/rest/JsonSerializerSuite.scala [hunk 1]
```diff
@@ -143,14 +187,68 @@ class JsonSerializerSuite extends BaseTestSuite with VersionedModelMatchers {
...
+        userUpdated = "system",
+
+        conformance = List(
+          CastingConformanceRule(0,
+            outputColumn = "ConformedInt",
+            controlCheckpoint = false,
+            inputColumn = "STRING_VAL",
+            outputDataType = "integer"
+          ),
+          MappingConformanceRule(1,
```
Comment (benedeki) [inline dao/src/test/scala/za/co/absa/enceladus/dao/rest/JsonSerializerSuite.scala:216]: **Minor:** Could you add another `MappingConformanceRule` without the field of `mappingTableFilter`, to prove backward compatibility.

**Review by benedeki — APPROVED** [review]
One little enhancement, if you will have time.

File: dao/src/test/scala/za/co/absa/enceladus/dao/rest/JsonSerializerSuite.scala (hunk 1 above)
Comment (dk1844) [reply dao/src/test/scala/za/co/absa/enceladus/dao/rest/JsonSerializerSuite.scala:229]: Good idea, thanks, added.

**Review by AdrianOlosutean — APPROVED** [review]

**Pull request merged**